heareval/predictions/task_predictions.py

```diff
@@ -781,13 +808,26 @@ def task_predictions_train(
     in_memory: bool,
     deterministic: bool,
 ) -> GridPointResult:
+    """
+    Train a predictor for a specific task using pre-computed embeddings.
+    """
+
     start = time.time()
     predictor: AbstractPredictionModel
     if metadata["embedding_type"] == "event":
-        validation_target_events = json.load(
-            embedding_path.joinpath("valid.json").open()
-        )
-        test_target_events = json.load(embedding_path.joinpath("test.json").open())
+
+        # Save the target events for validation and test
+        validation_target_events = {}
+        for split_name in data_splits["valid"]:
+            validation_target_events.update(
```

Comment:
Why are we doing an update? how do we make sure we are not clobbering /. overwriting existing keys? This is so confusing

Reply:
We are doing an update since the target event might now be coming from a set of splits